app/controllers/upvotes_controller.rb

```diff
@@ -0,0 +1,14 @@
+class UpvotesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    review = Review.find(params[:review_id])
+    if review.increment_vote(current_user)
+      flash[:success] = "Upvote saved."
+    else
+      flash[:alert] = "You have already upvoted."
+    end
+
+    redirect_to restaurant_path(review.restaurant)
+  end
+end
```

Comment:
You can probably merge the `UpvotesController` and `DownvotesController` and just pass in a 1 or -1 depending on whether it was an upvote or downvote.